aws/resource_aws_s3_bucket_test.go

```diff
@@ -60,17 +60,21 @@ func testSweepS3Buckets(region string) error {
 	for _, bucket := range output.Buckets {
 		name := aws.StringValue(bucket.Name)
 
-		hasPrefix := false
+		sweepable := false
 		prefixes := []string{"mybucket.", "mylogs.", "tf-acc", "tf-object-test", "tf-test", "tf-emr-bootstrap"}
 
 		for _, prefix := range prefixes {
 			if strings.HasPrefix(name, prefix) {
-				hasPrefix = true
+				sweepable = true
 				break
 			}
 		}
 
-		if !hasPrefix {
+		if matched, _ := regexp.MatchString(`^terraform-\d+$`, name); matched == true {
```

Comment:
Linting is unhappy about this line:

```
aws/resource_aws_s3_bucket_test.go:73:38: SA6000: calling regexp.MatchString in a loop has poor performance, consider using regexp.Compile (staticcheck)
		if matched, _ := regexp.MatchString(`^terraform-\d+$`, name); matched == true {
		                                   ^
```

The regular expression can be compiled above line 60 via something like:

```go
terraformRegexp := regexp.MustCompile(`^terraform-\d+$`)
```

And checked here via:

```go
if terraformRegexp.MatchString(name) {
```

Although it might just be easier to add `terraform-20` and `terraform-remote-s3-test-` to the prefixes.

Reply:
`terraform-remote-s3-test-` is generated by the remote backend, isn't it? Would it make sense for the cleanup to be in the backend tests?